### PR TITLE
Flash p gz bug

### PR DESCRIPTION
**Pull request by hepcat72**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/flash/flash.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
                     '$layout.reads.forward' '$layout.reads.reverse'
                     #set $input = $layout.reads.forward
                 #end if
-                #if $input.is_of_type('fastqsanger'):
+                #if $input.is_of_type('fastqsanger', 'fastqsanger.gz'):
```
Comment (mvdbeek) [inline tools/flash/flash.xml]: This translates to an `isinstance` check, so just `fastqsanger` should be enough here.

Comment (hepcat72) [reply tools/flash/flash.xml]: I did have just fastqsanger in the version Lance merged yesterday, but it was still supplying "-p 64”. After I added fastqsanger.gz, it supplies “-p 33”.

Comment (hepcat72) [reply tools/flash/flash.xml]: OK, yeah.  So I just tried changing it back to `#if $input.is_of_type('fastqsanger'):`, and it supplied `-p 64`:

```
flash --threads=${GALAXY_SLOTS:-1} -m 10 -M 65 -x 0.25  '/private/var/folders/8d/z84mms353qb2n0r3n2jhww0m0000gq/T/tmpKO5mvt/files/000/dataset_27.dat' '/private/var/folders/8d/z84mms353qb2n0r3n2jhww0m0000gq/T/tmpKO5mvt/files/000/dataset_28.dat' -p 64  > '/private/var/folders/8d/z84mms353qb2n0r3n2jhww0m0000gq/T/tmpKO5mvt/files/000/dataset_33.dat'
```

And the test fails with:

```
Fatal error: Exit code 1 ()
[FLASH] ERROR: Qual string contains character under phred_offset = 64 (file "/private/var/folders/8d/z84mms353qb2n0r3n2jhww0m0000gq/T/tmpKO5mvt/files/000/dataset_27.dat", near line 5)
[FLASH] ERROR: Qual string contains character under phred_offset = 64 (file "/private/var/folders/8d/z84mms353qb2n0r3n2jhww0m0000gq/T/tmpKO5mvt/files/000/dataset_28.dat", near line 5)
```

So I assume I have something wrong elsewhere?

**Comment by lparsons** [comment]
I believe you need to supply the `-z` flag if you want compressed output (which I personally think it should always output).

**Comment by hepcat72** [comment]
> I believe you need to supply the `-z` flag if you want compressed output (which I personally think it should always output).

I'd be happy if it just supplied the correct value for -p.  Do you have any idea why Marius's suggestion ends up resulting in `-p 64` being supplied?

**Comment by lparsons** [comment]
No, that would require some troubleshooting. However, if you're going through the trouble to update the tool, it would be very helpful to output compressed files.

**Comment by hepcat72** [comment]
Outputting gz would be nice, but it's moot if I can't get -p to set correctly.

**Comment by hepcat72** [comment]
OK.  I suspected this, but I didn't say because I wasn't sure.  It apparently outputs in gz if the input is gz.  I just changed the tests to expect fastqsanger (not fastqsanger.gz) and re-added the 'fastqsanger.gz' in the is_of_type call.  It fails because it's outputting gz:

```
JobOutputsError: Dataset metadata verification for [file_ext] failed, expected [fastqsanger] but found [fastqsanger.gz].
```

So it does output gz.  I just can't get -p 33 when I only have `#if $input.is_of_type('fastqsanger'):` which @mvdbeek says should work...

**Comment by mvdbeek** [comment]
There's no harm in specifying both `fastqsanger` and `fastqsanger.gz`, I'll add that just using fastqsanger doesn't work to the list of things to look into.

**Comment by lparsons** [comment]
@hepcat72 This needs a version bump and then it looks good to me.

**Comment by hepcat72** [comment]
I don't understand why the output test files (e.g. tools/flash/test-data/flash_unmerged_f_out4.fastqsanger) pass the tests, given they're not zipped. Is that a problem or does the test structure unzip the output files in order to compare them?

**Comment by hepcat72** [comment]
Oh, never mind.  My files changed tab was stale.

**Comment by hepcat72** [comment]
I think we're good to go now.

File: tools/flash/flash.xml [hunk 2]
```diff
@@ -146,6 +149,35 @@
             <output name="unmerged_reads_r" file="flash_unmerged_r_out3.fastqillumina" ftype="fastqillumina" />
             <output name="hist" file="flash_hist_out3.tabular" />
         </test>
+        <test>
+            <param name="select_layout" value="individual" />
+            <param name="forward" value="flash_forward_in4.fastqsanger.gz" ftype="fastqsanger.gz" />
+            <param name="reverse" value="flash_reverse_in4.fastqsanger.gz" ftype="fastqsanger.gz" />
+            <param name="generate_histogram" value="false" />
+            <param name="save_log" value="true" />
+            <output name="merged_paired_reads" file="flash_merged_out4.fastqsanger.gz" ftype="fastqsanger.gz" compare="sim_size" />
```
Comment (lparsons) [inline tools/flash/flash.xml]: Does anyone know what the default `delta` is for `sim_size`? I didn't see it in the docs.

**Comment by lparsons** [comment]
Looks good to me.

File: tools/flash/flash.xml (hunk 2 above)
Comment (hepcat72) [reply tools/flash/flash.xml]: Oh right.  I forgot you said a delta should be set...

**Pull request merged**